content/blog/2020-09-28-september-20-community-gems.md

```diff
@@ -0,0 +1,198 @@
+```
+
+To fix this, convert your RSA key to the OpenSSH format. Tools like PuTTYgen and
+MobaKeyGen can help you do this.
+
+### [Q: Can I have multiple `param.yaml` files in a project?](https://discordapp.com/channels/485586884165107732/563406153334128681/753322309942509578)
+
+Yes, you can have as many separate parameter files as you'd like. It's only
+important that they are correctly specified in your DVC pipeline stages.
+
+For example, if you have files `params_data_processing.yaml` and
+`params_model.yaml` in your project (perhaps to store hyperparameters of your
+data processing and model fitting stages, respectively), you'll want to call the
+right file at each stage. For example:
+
+```dvc
+dvc run -n preprocess \
+  -p params_data_process.yaml:param1,param2,...
+```
+
+### [Q: Is there a way to automatically produce SVG plots from `dvc plot`? I don't like having to click through the Vega-Lite GUI to get an SVG, and my plots look so small when I access them in the browser.](https://discordapp.com/channels/485586884165107732/563406153334128681/750012082149392414)
+
+If your DVC plots (and by DVC plots, we mean Vega-Lite plots 😉) look small in
+your browser, you can modify this programmatically! DVC generates Vega-Lite
+plots by way of a few templates that come pre-loaded. The templates are in
+`.dvc/plots` (assuming you're in a DVC directory).
+
+Find the template that corresponds to your plot (if you didn't specify a plot
+type in your CLI command, it's probably `default.json`) and modify the `height`
+and `width` paramters. Then save your changes.
+
+For more about how to modify your plot templates, check out the
+[Vega docs](https://vega.github.io/vega/docs/specification/). If you're
+considering making a whole new template that's custom for your data viz needs,
+[we've got docs on that](https://dvc.org/doc/command-reference/plots#custom-templates),
+too.
+
+One last tip: did you know about the
+[Vega-Lite CLI](https://anaconda.org/conda-forge/vega-lite-cli)? It provides
+functions for converting Vega-Lite plots to `.pdf`,`.png`,`.svg`, and `.vg`
+(Vega) formats. To use this approach with DVC, you'll want to use the
+`--show-vega` flag to print your plot specification to a `.json` file.
+
+```dvc
+$ dvc plots --show-vega  > vega.json
+$ vl2svg vega.json
+```
+
+### [Q: I'm confused about external dependencies and outputs. What's the difference?](https://discordapp.com/channels/485586884165107732/485596304961962003/752478399326453840)
+
+First, let's talk about **external outputs**. You might use an external output
+when you have a dataset in storage that is so large, you can't reasonably
+transfer it to your local workspace to start DVC-tracking it. For example, if
+you have a TB of data in S3 storage, you want to treat it as an external output
+so DVC can version it without resorting to time-consuming data transfer
+operations.
+
+External outputs are configured using using `dvc add --external`
```

Comment:
all good, except `dvc add` is not the only way ... `dvc run -o s3://something` also an external output.

In general, we can try to describe it here from a high level perspective- external outputs and dependencies are files, directories that you would `dvc add` or `dvc run -o` or `dvc run -d` but that physically reside outside of the local workspace (e.g. data is too large, or we are running a pipeline that writes directly to cloud, etc). To some extent I like to compare this with a "virtual workspace" or "expanded workspace". It just happened that some files that you add, produce, etc are located in some place outside.

We should mention that this is an advanced scenario. And we recommend to ask us on Discord for any help or confirm that this is needed, or if there are some performance bottlenecks.

Reply:
the question is about the difference between external outputs and external dependencies. i feel like it's not clear that they're very different, except in one potential case you need a cache (when you're versioning a remote dataset on cloud storage that you never move). 

do you think we should not write as if there's a difference between external outputs and dependencies? In other words, say they are the same thing?

Reply:
Terminology with outputs is pretty bad to mind. Output term comes from the pipelines and is being used even for a regular `dvc add something` while in this case from a user perspective this can be input to the pipeline, raw data. So, at the end of the day- they both represent a file or a directory somewhere outside that you don't want to copy locally to start using it. It can be `dvc add something` and we call it external output (😢 ), or it can be used in place of a dependency in `dvc run -d` - we call it external dependency then, or it can be an output/intermediate artifact of the pipeline `dvc run -o` - we call it external output as well.

From the cache/non-cache they behave exactly like regular local DVC-tracked directories, files and not-necessarily-DVC-tracked files and directories. For outputs you define cache and DVC puts into cache, for dependencies - we don't care - we just save hash to check if they changed.

So, I would they are not the same from a lot of angles. They are conceptually are very similar to regular files and directories that just happen to be located and cache outside.

Reply:
ahhh i see. yes it seems like the terminology should be reworked for clarity, eventually. but not the worst thing, we can work with it here.

Reply:
think i got it in the revision!